### PR TITLE
8205137: Remove Applet support from SwingSet2

### PR DESCRIPTION
**Pull request by alisenchung**
removed support for running demo from applet for J2Ddemo and SwingSet2
same PR as 8205137: Remove Applet support from SwingSet2 #5400 (changed branch name)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8205137](https://bugs.openjdk.java.net/browse/JDK-8205137): Remove Applet support from SwingSet2
 * [JDK-8205139](https://bugs.openjdk.java.net/browse/JDK-8205139): Remove Applet support from J2Ddemo

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**) ⚠️ Review applies to fe3bde46ca69e07bcf6290cd495f77ef9181ae0a
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**) ⚠️ Review applies to fe3bde46ca69e07bcf6290cd495f77ef9181ae0a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5401/head:pull/5401` \
`$ git checkout pull/5401`

Update a local copy of the PR: \
`$ git checkout pull/5401` \
`$ git pull https://git.openjdk.java.net/jdk pull/5401/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5401`

View PR using the GUI difftool: \
`$ git pr show -t 5401`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5401.diff">https://git.openjdk.java.net/jdk/pull/5401.diff</a>

</details>

### CODE REVIEW

**Comment by mrserb** [comment]
Probably the README in the "jdk/src/demo/share/" should be updated as well?
This block looks outdated:
```
In some cases, the default security settings may block an execution
of demo applets in a browser. To adjust the security settings, please
refer to the following resource:

http://java.com/en/download/help/java_blocked.xml

Some demo applets need to be accessed through the HTTP or HTTPS
protocols to enable access to the required resources.

```

**Comment by prrace** [comment]
The files that have JUST a (c) date update should all be reverted. You haven't touched them so there is nothing to (c).

**Comment by alisenchung** [comment]
> The files that have JUST a (c) date update should all be reverted. You haven't touched them so there is nothing to (c).

I just removed the java.applet.* import for all of them. Should I revert the date if there's no other content change?

**Comment by victordyakov** [comment]
@azuev-java @kizune please review

**Comment by prrace** [comment]
> > The files that have JUST a (c) date update should all be reverted. You haven't touched them so there is nothing to (c).
> 
> I just removed the java.applet.* import for all of them. Should I revert the date if there's no other content change?

Hmm .. I missed the delete.
Well .. what were those imports for ? Nothing used them.
Now, although arguably a separate issue, we have precedent for cleaning up imports at such a time.
There are LOTS of wild card imports. 
I'd like you to update this to current practice to explicitly enumerate the classes needed
so rather than 
import java.util.*;

import java.util.Map
etc ..
JUST those needed.

Then we may actually have something worthy of updating a (c) ..

**Comment by mrserb** [comment]
> Hmm .. I missed the delete.
> Well .. what were those imports for ? Nothing used them.
> Now, although arguably a separate issue, we have precedent for cleaning up imports at such a time.

This helped to prove that the usage of applets is actually removed from these files, I found it quite useful during the review.

**Comment by aivanov-jdk** [comment]
> > > The files that have JUST a (c) date update should all be reverted. You haven't touched them so there is nothing to (c).
> > 
> > 
> > I just removed the java.applet.* import for all of them. Should I revert the date if there's no other content change?
> 
> Hmm .. I missed the delete.
> Well .. what were those imports for ? Nothing used them.

It looks each file has the same wildcard import statements.

> Now, although arguably a separate issue, we have precedent for cleaning up imports at such a time.
> There are LOTS of wild card imports.

Many of the imports are unused. For example, `TreeDemo.java` doesn't use any of these: javax.swing.event.\*, javax.accessibility.\*, java.util.\*.

It makes sense to clean it up along with removing java.applet.* imports.

**Comment by aivanov-jdk** [comment]
> Probably the README in the "jdk/src/demo/share/" should be updated as well?

SwingSet2/README should also be updated.

File: src/demo/share/jfc/SwingSet2/HtmlDemo.java [hunk 1]
```diff
@@ -31,22 +31,19 @@
...
+import javax.swing.JViewport;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkEvent.EventType;
+import javax.swing.event.HyperlinkListener;
+import javax.swing.text.html.HTMLFrameHyperlinkEvent;
+import javax.swing.text.html.HTMLDocument;
 
-import java.awt.*;
-import java.awt.event.*;
-import java.beans.*;
```
Comment (prrace) [inline src/demo/share/jfc/SwingSet2/HtmlDemo.java:45]: So this is good clean up to have .. wildcard imports of so many packages that aren't even used !

**Review by prrace — APPROVED** [review]

**Review by aivanov-jdk — COMMENTED** [review]
In all the files except for `ButtonDemo.java`, you modify the imports only; in case of `ButtonDemo.java`, you also reformat the code. Is it intentional?

The imports in the files were in this order: `javax.*` followed by `java.*`. This order is preserved. However, in the JDK source files, imports from `java.*` usually come before `javax.*`.

What is the preferred order of imports?

**Comment by alisenchung** [comment]
> In all the files except for `ButtonDemo.java`, you modify the imports only; in case of `ButtonDemo.java`, you also reformat the code. Is it intentional?

Oh the reformat was unintentional, let me revert that...

**Review by azuev-java — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Comment by mrserb** [comment]
> What is the preferred order of imports?
Usually its is
java.*
empty line
javax.*
empty line
others.*
empty line 
static java.*
empty line
static javax.*
empty line
others.*

**Review by aivanov-jdk — APPROVED** [review]

File: src/demo/share/jfc/SwingSet2/DemoModule.java [hunk 2]
```diff
@@ -42,7 +42,6 @@
 import java.beans.*;
 import java.util.*;
 import java.io.*;
-import java.applet.*;
 import java.net.*;
```
Comment (prrace) [inline src/demo/share/jfc/SwingSet2/DemoModule.java:45]: Oops. I missed we still have wildcards here

File: src/demo/share/jfc/SwingSet2/TabbedPaneDemo.java [hunk 3]
```diff
@@ -44,7 +44,6 @@
 import java.beans.*;
 import java.util.*;
 import java.io.*;
-import java.applet.*;
 import java.net.*;
```
Comment (prrace) [inline src/demo/share/jfc/SwingSet2/TabbedPaneDemo.java:47]: And also stilll have them here.

**Review by prrace — COMMENTED** [review]
https://www.oracle.com/technetwork/java/codeconventions-150003.pdf didn't say anything relevant.

The still-in-draft update to this  http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-import-statements
which is focused on OpenJDK developers says

Import statements should be sorted…
…primarily by non-static / static with non-static imports first.
…secondarily by package origin according to the following order
java packages
javax packages
external packages (e.g. org.xml)
internal packages (e.g. com.sun)
…tertiary by package and class identifier in lexicographical order

So that isn't yet "blessed" and could be written a bit more clearly but I could also imagine quibbles these days
like "group imports by module, starting with java.base" as the primary one.

And imports from named modules go first ..

And I can imagine folks wanting to place exception imports after other imports ..

So I don't think there is any agreed upon set of rules to follow although most folks
when you get down to the specific package do place them in lexical order

Here I note that the pre-existing wild card imports were of javax.swing followed  by java.awt*
and the change is preserving that so I don't have a strong opinion on that nor do I think I
would insist on lexical ordering but I also would be quite fine if the fixer is OK to make that change.

**Comment by aivanov-jdk** [comment]
I asked about imports because I usually see `java.*` followed by `javax.*` followed by internal imports in the JDK code. I wanted to confirm my understanding.

If I remember correctly, IDEA has a different order by default: other imports, then `java.*` and `javax.*`. I can see the motivation here: for projects using Java, you care about your own classes more than `java(x)` packages. For JDK, java(x) are the important ones.

In this case, the previous order of the imports was preserved where `javax.*` imports were before `java.*`. In addition to that, SwingSet2 is a demo of Swing components, so having `javax.swing.*` above `java.awt.*` makes sense to me too.

> Here I note that the pre-existing wild card imports were of javax.swing followed by java.awt*
> and the change is preserving that so I don't have a strong opinion on that nor do I think I
> would insist on lexical ordering but I also would be quite fine if the fixer is OK to make that change.

I'm also fine with either order of the imports.

**Comment by mrserb** [comment]
I suggest using the same order no need to think about which order is correct in which file, but simple as this: https://github.com/openjdk/jdk/pull/5401#issuecomment-917305195

**Comment by prrace** [comment]
> I suggest using the same order no need to think about which order is correct in which file, but simple as this: [#5401 (comment)](https://github.com/openjdk/jdk/pull/5401#issuecomment-917305195)

My point is that you can't point to any doc that says "this is the order to be used" because we've never codified it.
So as long as it is reasonable .. ie not jumping from package to package, it's all good.

FWIW the old style guideline doesn't even have anything to say about not using wildcards.  I guess they were thought to be cool but as things got bigger it became vulnerable to clashes, so there is "general" acceptance that there's a good reason for explicit imports beyond aesthetics or style.

**Comment by mrserb** [comment]
> > I suggest using the same order no need to think about which order is correct in which file, but simple as this: [#5401 (comment)](https://github.com/openjdk/jdk/pull/5401#issuecomment-917305195)

> My point is that you can't point to any doc that says "this is the order to be used" because we've never codified it.

We did that, we discussed it someday long ago and I followed it for most of my code cleanups in java desktop module. It also was summarised here http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-import-statements
But yes we did not strictly follow it, not a big issue.

**Pull request closed without merging**